File: XMLCheck/Handler/CheckRulesXML07/Xml7RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML07
{
    public abstract class Xml7RuleBase:IXml7Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml7Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml7/xml7RuleBase.ts
import { Xml7Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';

export interface IXml7Rule {
  readonly key: string;
  check(model: Xml7Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml7RuleBase implements IXml7Rule {
  public abstract get key(): string;
  public abstract check(model: Xml7Model, context: HoSoContext): ErrorDetails | null;

  protected error(message: string): ErrorDetails {
    return {
      key: this.key,
      message,
    };
  }
}
